test/manifold_fuzz.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright 2024 The Manifold Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <atomic>
+#include <future>
+
+#include "fuzztest/fuzztest.h"
+#include "gtest/gtest.h"
+#include "manifold/manifold.h"
+
+using namespace fuzztest;
+
+enum class Transform { Translate, Rotate, Scale };
+
+// larger numbers may cause precision issues, prefer to test them later
+auto GoodNumbers = OneOf(InRange(0.1, 10.0), InRange(-10.0, -0.1));
+auto Vec3Domain = ArrayOf<3>(GoodNumbers);
+auto TransformDomain = PairOf(
+    ElementOf({Transform::Translate, Transform::Rotate, Transform::Scale}),
+    Vec3Domain);
+auto CsgDomain = VectorOf(PairOf(VectorOf(TransformDomain).WithMaxSize(20),
+                                 ElementOf({0, 1})))
+                     .WithMaxSize(100);
+
+void SimpleCube(
+    const std::vector<std::pair<
+        std::vector<std::pair<Transform, std::array<double, 3>>>, int>>
+        &inputs) {
```

Comment:
This is a bit hard to read - maybe a vector of structs?

Reply:
sure